### PR TITLE
Error message propagate to console when conda cannot find environments

### PR DESCRIPTION
**Pull request by jgutman**
### Intent

Addresses #10103.

### Approach
Take the error message from conda system command and send to console via stop. They might also hit this error via the UI (via Tools > Global Options > Python > Select python interpreter), so it could be worth showing the error in the UI as well instead of just
image

### Automated Tests

NA

### QA Notes

Would be best to have a way to trigger an error upon running `conda env list --json --quiet` in the shell in order to be able to properly test the error handling

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [X] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [X] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [X] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/session/modules/SessionPythonEnvironments.R [hunk 1]
```diff
@@ -639,7 +639,7 @@
    if (attr(output, "status") != 0) {
      errors <- paste(readLines(tmp), collapse = "\n")
      
-     stop(conda, args, ":\n", errors)
+     .rs.stopf("Error while executing %s %s:\n%s", conda, paste(args, collapse=" "), errors)
```
Comment (kevinushey) [inline src/cpp/session/modules/SessionPythonEnvironments.R:642]:
```suggestion
     .rs.stopf("Error executing %s %s:\n%s", conda, paste(args, collapse = " "), errors)
```

**Review by kevinushey — APPROVED** [review]
LGTM!

**Comment by jgutman** [comment]
> LGTM!

Do we want to get this error back to the UI? Is there a way to do that?

**Comment by kevinushey** [comment]
Good question... I had thought it might be available on the error object being logged here:

https://github.com/rstudio/rstudio/blob/cedab12df1b312a0ca1301b3442412c0c776d6a9/src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/PythonPreferencesPaneBase.java#L111-L120

Our R RPC methods get invoked here:

https://github.com/rstudio/rstudio/blob/cedab12df1b312a0ca1301b3442412c0c776d6a9/src/cpp/r/RJsonRpc.cpp#L126-L139

So if the error message is missing, I wonder if there's an appropriate place to tag it on the generated error object.

**Comment by kevinushey** [comment]
Or, I wonder if part of the issue is that this is just a warning, not an error, so it doesn't fail the RPC call? E.g. I see in the console:

```
Warning in system2(conda, args, stdout = TRUE, stderr = FALSE) :
  running command ''/opt/local/bin/conda' env list --json --quiet 2>/dev/null' had status 1
```

and I don't think that causes the rest of the code execution in this RPC to halt. Maybe that's fine?

**Comment by kevinushey** [comment]
> But that's the old code. I changed it to actually raise an error with stop (.rs.stopf)

Sorry, you're right. Not enough sleep / coffee last night...

Let me take a closer look.

**Comment by kevinushey** [comment]
It looks like we do not tag our R code execution errors with the underlying R error message, which is unfortunate:

Screen Shot 2022-01-26 at 11 58 15 AM

Getting that to work properly is going to be a larger change, I think.

**Comment by jgutman** [comment]
Okay. It's not too bad, the dialog box will be unhelpful, but then when the user closes the dialog box, they'll see the actual error message in their console. But seems like we should maybe make an issue for that, because it seems like at some point being able to pass, or categorize, error messages from R to the RPC could be helpful

**Comment by kevinushey** [comment]
> Okay. It's not too bad, the dialog box will be unhelpful, but then when the user closes the dialog box, they'll see the actual error message in their console. But seems like we should maybe make an issue for that, because it seems like at some point being able to pass, or categorize, error messages from R to the RPC could be helpful

Would you mind filing an issue for that?

**Comment by jgutman** [comment]
Note: to reproduce the error for further testing, navigate to `PYTHON_PREFIX/lib/pythonX.Y/site-packages/conda/_vendor/toolz/` and rename `itertoolz.py` to `XXXitertoolz.py`. You should then see an error when running `conda env list --json --quiet`. Be sure to rename it back when you're done.

**Pull request merged**